examples/ffi/exporter.cpp

```diff
@@ -120,18 +120,24 @@ int main(int argc, char *argv[]) {
 
   auto exporter = exporter_new_result.ok;
 
-  ddog_prof_Exporter_File files_[] = {{
+  ddog_prof_Exporter_File files_to_compress_and_export_[] = {{
       .name = DDOG_CHARSLICE_C("auto.pprof"),
       .file = ddog_Vec_U8_as_slice(&encoded_profile->buffer),
   }};
-
-  ddog_prof_Exporter_Slice_File files = {.ptr = files_, .len = sizeof files_ / sizeof *files_};
+  ddog_prof_Exporter_Slice_File files_to_compress_and_export = {.ptr = files_to_compress_and_export_, .len = sizeof files_to_compress_and_export_ / sizeof *files_to_compress_and_export_};
+  
+  // Ensure we get a non-null pointer: it is required that:
+  // "data must be non-null and aligned even for zero-length slices."
+  // https://doc.rust-lang.org/std/slice/fn.from_raw_parts.html
+  ddog_prof_Exporter_File files_to_export_unmodified_[1];
+  ddog_prof_Exporter_Slice_File files_to_export_unmodified = {.ptr = files_to_export_unmodified_, .len = 0};
```

Comment:
Hmmmmmmmmmmmmm.... can we do better for ffi consumers? Having to do this if you don't have any extra files is... painful.

(Alternatively, maybe we could `#define` a macro `NO_FILES` that ffi consumers can use for this case in the headers generated by cbindgen?)

To be clear -- my concern is not on about this example, it's more that I believe some of our profilers may need to do something similar.

Reply:
I'm going to ask @sanchda to back me up here, but this should be fine:

```cpp
ddog_prof_Exporter_Slice_File files_to_export_unmodified = {
    .ptr = &(ddog_prof_Exporter_File){},
    .len = 0,
};
```

Even if that address `&(ddog_prof_Exporter_File){}` becomes a dangling pointer, it doesn't matter: it will never be dereferenced, but it will still be properly aligned for a `ddog_prof_Exporter_File`. Right, David?

Reply:
Right; the struct you've defined is going to have a pointer to a default-constructed `ddog_prof_Exporter_File` on the stack.  It'll be a valid pointer for the duration of the current scope, after which it'll become invalid.  The requirement that `.ptr` be "valid" is a little painful, but that's imposed on us.

I like @ivoanjo's note; create a macro (equivalently, just a global pointer to a static-initialized element) that references a static singleton element with 0 .len and a payload that contains some kind of clear error message or whatever, then pass that.

For discussion, I have a vague sense that the standard might be a little overbearing here; this is a fairly simple struct, so maybe just passing `{}` will be acceptable (I didn't test it or think too hard about this).  I'd argue that _if_ this is viable, then wrapping it in a macro anyway is better, and thus just going through the trouble of having a macro pass a sentinel zero-file is better.

I don't really have a strong opinion.  I'd wrap this in a helper function on my side anyway if I had to do this in more than one place.  That said, I sense @ivoanjo's comment was less about the unbearable burden of doing this once and more about having to follow _yet another thing_ across API changes as the library evolves.  :shrug:

Reply:
I'm also fine with making `ddog_prof_Exporter_Slice_File_empty()` which returns an empty `ddog_prof_Exporter_Slice_File` for you.

Reply:
`ddog_prof_Exporter_Slice_File_empty()` is quite reasonable, happy to go with that :)